api/src/context.rs

```diff
@@ -91,9 +91,15 @@ impl Context {
     }
 
     pub fn get_latest_ledger_info(&self) -> Result<LedgerInfo, Error> {
+        let oldest_version = self.db.get_first_txn_version()?;
+
+        if oldest_version.is_none() {
+            return Err(anyhow! {"Failed to retrieve oldest version"}.into());
+        }
```

Comment:
rather than introducing an unwrap, can we use some `if let some(...) ... else return Err...?

Reply:
Yeah, there's a ton of unwraps necessary, so I was inconsistent with my approach.  Will clean up.